general/kdtree.hpp

```diff
@@ -635,6 +637,16 @@ class KDTree : public KDTreeBase<Tindex, Tfloat>
    }
 };
 
+#ifdef MFEM_USE_SINGLE
+/// Defines KDTree in 3D
+typedef KDTree<int,float,3> KDTree3D;
+
+/// Defines KDTree in 2D
+typedef KDTree<int,float,2> KDTree2D;
+
+/// Defines KDTree in 1D
+typedef KDTree<int,float,1> KDTree1D;
+#elif defined MFEM_USE_DOUBLE
```

Comment:
```suggestion
using KDTree1D = KDTree<int,real_t,1>;
using KDTree2D = KDTree<int,real_t,2>;
using KDTree3D = KDTree<int,real_t,3>;
```

(and delete the `double` typedefs below). Will need `#include "globals.hpp"` in this header too. (Perhaps `real_t` should be defined in `config.hpp` rather than `globals.hpp`?)

Reply:
Thanks @pazner, I put the `KDTree{N}D` back, with this suggestion, and no longer have CI errors for this.